mobile/lib/services/deep_link.service.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:immich_mobile/providers/album/current_album.provider.dart';
+import 'package:immich_mobile/providers/asset_viewer/current_asset.provider.dart';
+import 'package:immich_mobile/routing/router.dart';
+import 'package:immich_mobile/services/album.service.dart';
+import 'package:immich_mobile/services/asset.service.dart';
+import 'package:immich_mobile/services/memory.service.dart';
+import 'package:immich_mobile/widgets/asset_grid/asset_grid_data_structure.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+final deepLinkServiceProvider = Provider(
+  (ref) => DeepLinkService(
+    ref.watch(memoryServiceProvider),
+    ref.watch(assetServiceProvider),
+    ref.watch(albumServiceProvider),
+    ref.watch(currentAssetProvider.notifier),
+    ref.watch(currentAlbumProvider.notifier),
+  ),
+);
+
+class DeepLinkService {
+  final MemoryService _memoryService;
+  final AssetService _assetService;
+  final AlbumService _albumService;
+  final CurrentAsset _currentAsset;
+  final CurrentAlbum _currentAlbum;
+
+  DeepLinkService(
+    this._memoryService,
+    this._assetService,
+    this._albumService,
+    this._currentAsset,
+    this._currentAlbum,
+  );
+
+  Future<DeepLink> handleScheme(PlatformDeepLink link, bool isColdStart) async {
+    // get everything after the scheme, since Uri cannot parse path
+    final intent = link.uri.host;
+    final queryParams = link.uri.queryParameters;
+
+    PageRouteInfo<dynamic>? deepLinkRoute;
+
+    switch (intent) {
+      case "memory":
+        deepLinkRoute = await _buildMemoryDeepLink(queryParams['id'] ?? '');
+      case "asset":
+        deepLinkRoute = await _buildAssetDeepLink(queryParams['id'] ?? '');
+      case "album":
+        deepLinkRoute = await _buildAlbumDeepLink(queryParams['id'] ?? '');
+    }
+
+    // Deep link resolution failed, safely handle it based on the app state
+    if (deepLinkRoute == null) {
+      if (isColdStart) return DeepLink.defaultPath;
+      return DeepLink.none;
+    }
+
+    return DeepLink([
+      // we need something to segue back to if the app was cold started
+      if (isColdStart) const PhotosRoute(),
```

Comment:
avoid using one liner, we try to put everthing into the curly braces

Reply:
I can't do that here because its a ternary in an array. See https://github.com/immich-app/immich/blob/ac49eef98a4da41cfacbdfbfbe804d60e36f43f4/mobile/lib/widgets/asset_viewer/top_control_app_bar.dart#L206